## Present Value of a Purchase

I am buying a car today for $30000. Ho much it actually costs me.

In [1]:
P = 30000
n = 10 # years
r = 0.1 # market rate of return
i = 0.03 # inflation


In [4]:
import numpy as np
import numpy_financial as npf

In [9]:
FV = npf.fv(r,n,0,P)
FV

np.float64(-77812.27380300008)

In [12]:
PV = npf.pv(i,n,0,FV)
PV

np.float64(57899.6394410902)

In [14]:
PV/P

np.float64(1.9299879813696734)

Actual value of the purchase is 58000. Almost double (1.93 times) the initial investment.

### With Fixed Periodic Maintenance Cost

Now let's assume that the car needs $1000 additional cost per year (maintenance, insurance etc.)

For the simple approach we assume that $1000 cost is fixed amount.

In [53]:
P = 30000
p = 1000
n = 10 # years
r = 0.1 # market rate of return
i = 0.03 # inflation

In [54]:
# Funture value of P:
FVP = P*(1+r)**n
FVP

77812.27380300008

In [55]:
# Future value of regular payments p:
FVp = p*((1+r)**n-1)/r
FVp

15937.424601000022

In [56]:
# Present value
PV = (FVP + FVp)/(1+i)**n
PV

69758.58010581968

Taking in account adds 12000 more (40%) to the total cost of the purchase.

### With Growing Periodic Maintenance Cost

In [24]:
P = 30000
p = 1000
n = 10 # years
r = 0.1 # market rate of return
i = 0.03 # inflation

In [46]:
# Funture value of P:
FVP = P*(1+r)**n
FVP

77812.27380300008

In [48]:
# Actual inflated periodic payments
period_index = np.arange(10)
actual_p = p*np.pow(1+i, period_index)
actual_p

array([1000.        , 1030.        , 1060.9       , 1092.727     ,
       1125.50881   , 1159.2740743 , 1194.05229653, 1229.87386542,
       1266.77008139, 1304.77318383])

In [50]:
# Future value of actual periodic payments
reversed_period_index = period_index[::-1]
FVp = sum(actual_p*(1+r)**reversed_period_index)
FVp

np.float64(17854.658296512556)

In [52]:
# Present value
PV_inflated = (FVP + FVp)/(1+i)**n
PV_inflated

np.float64(71185.18203208552)

In [57]:
PV_inflated - PV

np.float64(1426.6019262658374)

In [58]:
(PV_inflated - PV)/P*100

np.float64(4.755339754219458)

In [60]:
round((PV_inflated - P)/P*100, 2)

np.float64(137.28)

Taking in account inflation adds another $1400 (about 4.8%).

The actual value of our car purchase is $71185 (137.28% more than current price).

### Generalization

$$
PV_{price} = P \times \left(\frac{1+r}{1+i}\right)^n
$$

$$
{actual\ payments} = p,\ p(1+i),\ ...,\ p(1+i)^{(n-1)}
$$

$$
FV_{actual\ payments} = p(1+r)^{(n-1)} + p(1+r)^{(n-2)}(1+i) + ... + p(1+i)^{(n-1)}
$$

$$
PV_{actual\ payments} = \frac{FV_{actual\ payments}}{(1+i)^n}
$$


In [63]:
n = 10
r = 0.1
i = 0.03
periods = np.arange(n)
reverse_periods = periods[::-1]

In [72]:
price_factor = (1+r)**n/(1+i)**n
price_factor = round(price_factor, 2)
price_factor

1.93

In [71]:
payment_factor = ((1+i)**periods*(1+r)**reverse_periods).sum()/(1+i)**n
payment_factor = round(payment_factor, 2)
payment_factor

np.float64(13.29)

In [73]:
30000*price_factor + 1000*payment_factor

np.float64(71190.0)